viewers/nerf_viewer/bridge_server/bridge_server_subprocess.py

```diff
@@ -0,0 +1,85 @@
+import atexit
+import os
+import signal
+import socket
+import subprocess
+import sys
+import threading
+import time
+
+import server
+
+from typing import Optional
+
+
+def is_port_open(port: int):
+    # check whether the given port is open
+    try:
+        sock = socket.socket()
+        sock.bind(("", port))
+        sock.close()
+
+        return True
+    except OSError:
+        return False
+
+
+def get_free_port(default_port: int = None):
```

Comment:
```suggestion
from typing import Union
def get_free_port(default_port: Union[int, None] = None):
```

Reply:
The `default_port: int` has been declared as `default_port: Union[int, None]`.